DOL.WHD.Section14c.Business/Services/AttachmentService.cs

```diff
@@ -16,27 +18,33 @@ public class AttachmentService : IAttachmentService
         private readonly IFileRepository _fileRepository;
         private readonly IAttachmentRepository _attachmentRepository;
         private bool Disposed = false;
+        private readonly string FileEncryptKey = "@WSX3edc$RFV5tgb^YHN7ujm";
 
         public AttachmentService(IFileRepository fileRepository, IAttachmentRepository attachmentRepository)
         {
             _fileRepository = fileRepository;
             _attachmentRepository = attachmentRepository;
         }
 
-        public Attachment UploadAttachment(string EIN, byte[] bytes, string fileName, string fileType)
+        public Attachment UploadAttachment(string applicationId, byte[] bytes, string fileName, string fileType)
         {
             var fileUpload = new Attachment()
             {
                 FileSize = bytes.Length,
                 MimeType = fileType,
                 OriginalFileName = fileName,
                 Deleted = false,
-                ApplicationId = EIN
+                ApplicationId = applicationId
             };
 
-            fileUpload.RepositoryFilePath = $@"{EIN}\{fileUpload.Id}";
+            fileUpload.RepositoryFilePath = $@"{applicationId}\{fileUpload.Id}";
 
-            _fileRepository.Upload(bytes, fileUpload.RepositoryFilePath);
+            // Encrypt file         
+            byte[] keyInBytes = Encoding.UTF8.GetBytes(FileEncryptKey);
+            // Hash the password with SHA256
+            keyInBytes = SHA256.Create().ComputeHash(keyInBytes);
```

Comment:
One quick way to have unique keys per file would be to have a salt here, where the salt is something intrinsic to the file (its filename, for example).  E.g., `SHA256.Create().ComputeHash(keyInBytes + fileNameInBytes);`

Reply:
Done